site/app/entities/forum/Post.php

```diff
@@ -63,9 +104,80 @@ class Post {
     #[ORM\Column(type: Types::BOOLEAN)]
     protected bool $render_markdown;
 
+    public function isRenderMarkdown(): bool {
+        return $this->render_markdown;
+    }
+
     /**
      * @var Collection<PostHistory>
      */
     #[ORM\OneToMany(mappedBy: "post", targetEntity: PostHistory::class)]
     protected Collection $history;
+
+    /**
+     * @return Collection<PostHistory>
+     */
+    public function getHistory(): Collection {
+        return $this->history;
+    }
+
+    /**
+     * @var Collection<PostAttachment>
+     */
+    #[ORM\OneToMany(mappedBy: "post", targetEntity: PostAttachment::class)]
+    protected Collection $attachments;
+
+    /**
+     * @return Collection<PostAttachment>
+     */
+    public function getAttachments(): Collection {
+        return $this->attachments;
+    }
+
+    /**
+     * @var Collection<UserEntity>
+     */
+    #[ORM\ManyToMany(targetEntity: UserEntity::class, inversedBy: "upducks")]
+    #[ORM\JoinTable(name: "forum_upducks")]
+    #[ORM\JoinColumn(name: "post_id", referencedColumnName:"id")]
+    #[ORM\InverseJoinColumn(name:"user_id", referencedColumnName:"user_id")]
+    protected Collection $upduckers;
+
+    /**
+     * @return Collection<UserEntity>
+     */
+    public function getUpduckers(): Collection {
+        return $this->upduckers;
+    }
+
+    protected int $reply_level = 1;
+    public function getReplyLevel(): int {
+        return $this->reply_level;
+    }
+    public function setReplyLevel(int $new): void {
+        $this->reply_level = $new;
+    }
```

Comment:
Is this allowed to be negative?

Reply:
It's not, but this value also doesn't persist. If this gets set negative it'll either cause a server error or render weird, but I don't know how much it matters whether the request crashes here or later down the line.